lib/dao.js

```diff
@@ -285,21 +285,39 @@ DataAccessObject.create = function (data, options, cb) {
           }
           obj.__persisted = true;
 
-          saveDone.call(obj, function () {
-            createDone.call(obj, function () {
-              if (err) {
-                return cb(err, obj);
-              }
-              var context = {
-                Model: Model,
-                instance: obj,
-                isNewInstance: true,
-                hookState: hookState,
-                options: options
-              };
-              Model.notifyObserversOf('after save', context, function(err) {
-                cb(err, obj);
-                if(!err) Model.emit('changed', obj);
+          var context = {
+            Model: Model,
+            data: val,
+            isNewInstance: true,
+            hookState: hookState,
+            options: options
+          };
+          Model.notifyObserversOf('loaded', context, function(err) {
+
+            // Per current implementation, context of `create` does NOT 
+            // have the data updated through persist/loaded hooks.  
+            // So, introduced a new property `updateOnLoad`, which 
+            // if set, returns the updated context.
```

Comment:
I find this comment mildly confusing, could you please reword it her and also in other places where it's copied?

 > context of `create` does NOT have the data updated through persist/loaded hooks

As I understand it, the context DOES have the updated data. It's the instance passed to the callback which does not have the updated data.

How about the following version? (Feel free to reword it to your taste.)

By default, the instance passed to `create` callback is NOT updated with the changes made through persist/loaded hooks. To preserve backwards compatibility, we introduced a new setting `updateOnLoad`, which if set, will apply these changes to the model instance too.